ktor-client/ktor-client-cio/jvm/src/io/ktor/client/engine/cio/Platform.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2014-2023 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.engine.cio
+
+import java.lang.reflect.InvocationTargetException
+
+internal actual object Platform {
+    /**
+     * This explicit check avoids activating in Android Studio with Android specific classes available when running plugins inside the IDE.
+     */
+    private val isAndroid: Boolean = System.getProperty("java.vm.name") == "Dalvik"
+
+    actual fun isCleartextTrafficPermitted(hostname: String): Boolean {
```

Comment:
These aren't the full story for OkHttp.

We compile against robolectric android-all in order to call these APIs directly.

https://github.com/square/okhttp/blob/okhttp_4x/okhttp/src/main/kotlin/okhttp3/internal/platform/AndroidPlatform.kt

It's way simpler than the reflection.

Reply:
Yeah, I borrowed these code from the old implementation, cause I see there is no Android Runtime applied in CIO module.

Reply:
Same for OkHttp. It's a compile only dependency